stock_resupply_order/models/stock_resupply_order.py

```diff
@@ -0,0 +1,191 @@
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class StockResupplyOrder(models.Model):
+    _name = "stock.resupply.order"
+    _inherit = ["mail.thread"]
+    _description = """
+    Creates resupply orders that create procurements
+    following stocks that are already in the destination location
+    """
+
+    location_id = fields.Many2one(
+        "stock.location",
+        required=True,
+        help="The stock location that needs to be restocked",
+    )
+
+    company_id = fields.Many2one(
+        "res.company",
+        "Company",
+        required=True,
+        index=True,
+        default=lambda self: self.env.company,
+    )
+
+    stock_resupply_order_lines = fields.One2many(
```

Comment:
to follow common practices I would have call this field differently

```suggestion
    stock_resupply_order_line_ids = fields.One2many(
```